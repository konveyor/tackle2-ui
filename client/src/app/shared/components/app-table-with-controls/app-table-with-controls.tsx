import React from "react";
import { useTranslation } from "react-i18next";

import {
  Toolbar,
  ToolbarContent,
  ToolbarItem,
  ToolbarItemVariant,
} from "@patternfly/react-core";

import { AppTable, IAppTableProps } from "../app-table/app-table";
import { PaginationStateProps } from "@app/shared/hooks/useLegacyPaginationState";
import { SimplePagination } from "../simple-pagination";

export interface IAppTableWithControlsProps extends IAppTableProps {
  count: number;
  withoutTopPagination?: boolean;
  withoutBottomPagination?: boolean;
  toolbarBulkSelector?: any;
  toolbarToggle?: any;
  toolbarActions?: any;
  toolbarClearAllFilters?: () => void;
  paginationProps: PaginationStateProps;
  paginationIdPrefix?: string;
}

export const AppTableWithControls: React.FC<IAppTableWithControlsProps> = ({
  count,
  withoutTopPagination,
  withoutBottomPagination,
  toolbarBulkSelector,
  toolbarToggle,
  toolbarActions,
  toolbarClearAllFilters,
  paginationProps,
  paginationIdPrefix,
  ...rest
}) => {
  const { t } = useTranslation();

  return (
    <div
      style={{
        backgroundColor: "var(--pf-v5-global--BackgroundColor--100)",
      }}
    >
      <Toolbar
        className="pf-m-toggle-group-container"
        collapseListedFiltersBreakpoint="xl"
        clearAllFilters={toolbarClearAllFilters}
        clearFiltersButtonText={t("actions.clearAllFilters")}
      >
        <ToolbarContent>
          {toolbarBulkSelector}
          {toolbarToggle ? toolbarToggle : null}
          {toolbarActions}
          {!withoutTopPagination && (
            <ToolbarItem
              variant={ToolbarItemVariant.pagination}
              align={{ default: "alignRight" }}
            >
              <SimplePagination
                idPrefix={paginationIdPrefix}
                isTop={true}
                paginationProps={paginationProps}
              />
            </ToolbarItem>
          )}
        </ToolbarContent>
      </Toolbar>
      <AppTable {...rest} />
      {!withoutBottomPagination && (
        <SimplePagination
          idPrefix={paginationIdPrefix}
          isTop={false}
          paginationProps={paginationProps}
        />
      )}
    </div>
  );
};
