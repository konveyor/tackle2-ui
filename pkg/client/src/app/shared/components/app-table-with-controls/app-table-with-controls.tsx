import React from "react";
import { useTranslation } from "react-i18next";

import {
  Toolbar,
  ToolbarContent,
  ToolbarItem,
  ToolbarItemVariant,
  ToolbarToggleGroup,
} from "@patternfly/react-core";

import { FilterIcon } from "@patternfly/react-icons/dist/esm/icons/filter-icon";

import { AppTable, IAppTableProps } from "../app-table/app-table";
import { SimplePagination } from "../simple-pagination";

export interface IAppTableWithControlsProps extends IAppTableProps {
  count: number;
  pagination: {
    perPage?: number;
    page?: number;
  };
  onPaginationChange: ({
    page,
    perPage,
  }: {
    page: number;
    perPage: number;
  }) => void;

  withoutTopPagination?: boolean;
  withoutBottomPagination?: boolean;

  toolbarBulkSelector?: any;
  toolbarToggle?: any;
  toolbarActions?: any;
  toolbarClearAllFilters?: () => void;
}

export const AppTableWithControls: React.FC<IAppTableWithControlsProps> = ({
  count,
  pagination,
  onPaginationChange,

  withoutTopPagination,
  withoutBottomPagination,

  toolbarBulkSelector,
  toolbarToggle,
  toolbarActions,
  toolbarClearAllFilters,

  ...rest
}) => {
  const { t } = useTranslation();

  return (
    <div style={{ backgroundColor: "var(--pf-global--BackgroundColor--100)" }}>
      <Toolbar
        className="pf-m-toggle-group-container"
        collapseListedFiltersBreakpoint="xl"
        clearAllFilters={toolbarClearAllFilters}
        clearFiltersButtonText={t("actions.clearAllFilters")}
      >
        <ToolbarContent>
          {toolbarBulkSelector}
          {toolbarToggle && (
            <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">
              {toolbarToggle}
            </ToolbarToggleGroup>
          )}
          {toolbarActions}
          {!withoutTopPagination && (
            <ToolbarItem
              variant={ToolbarItemVariant.pagination}
              alignment={{ default: "alignRight" }}
            >
              <SimplePagination
                count={count}
                params={pagination}
                onChange={onPaginationChange}
                isTop={true}
              />
            </ToolbarItem>
          )}
        </ToolbarContent>
      </Toolbar>
      <AppTable {...rest} />
      {!withoutBottomPagination && (
        <SimplePagination
          count={count}
          params={pagination}
          onChange={onPaginationChange}
        />
      )}
    </div>
  );
};
