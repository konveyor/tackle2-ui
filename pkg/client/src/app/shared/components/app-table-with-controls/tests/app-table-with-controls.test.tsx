import React from "react";
import { mount, shallow } from "enzyme";
import { ICell } from "@patternfly/react-table";

import { AppTableWithControls } from "../app-table-with-controls";

describe("AppTableWithControls", () => {
  const columns: ICell[] = [{ title: "Col1" }];
  const itemsToRow = (items: string[]) => {
    return items.map((item) => ({
      cells: [
        {
          title: item,
        },
      ],
    }));
  };

  it("Renders without crashing", () => {
    const wrapper = shallow(
      <AppTableWithControls
        count={120}
        rows={[{ cells: ["first", "second", "third"] }]}
        pagination={{ page: 1, perPage: 10 }}
        onPaginationChange={jest.fn()}
        onSort={jest.fn()}
        cells={columns}
        isLoading={false}
        filtersApplied={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders adding toolbar", () => {
    const wrapper = shallow(
      <AppTableWithControls
        count={120}
        rows={[{ cells: ["first", "second", "third"] }]}
        pagination={{ page: 1, perPage: 10 }}
        onPaginationChange={jest.fn()}
        onSort={jest.fn()}
        cells={columns}
        isLoading={false}
        toolbarActions={<p>This is an additional content to toolbar</p>}
        filtersApplied={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
