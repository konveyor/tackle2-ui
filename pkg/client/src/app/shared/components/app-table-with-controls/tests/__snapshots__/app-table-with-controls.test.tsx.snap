// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTableWithControls Renders adding toolbar 1`] = `
<div
  style={
    Object {
      "backgroundColor": "var(--pf-global--BackgroundColor--100)",
    }
  }
>
  <Toolbar
    className="pf-m-toggle-group-container"
    clearFiltersButtonText="actions.clearAllFilters"
    collapseListedFiltersBreakpoint="xl"
  >
    <ToolbarContent
      isExpanded={false}
      showClearFiltersButton={false}
    >
      <p>
        This is an additional content to toolbar
      </p>
      <ToolbarItem
        alignment={
          Object {
            "default": "alignRight",
          }
        }
        variant="pagination"
      >
        <SimplePagination
          count={120}
          isTop={true}
          onChange={[MockFunction]}
          params={
            Object {
              "page": 1,
              "perPage": 10,
            }
          }
        />
      </ToolbarItem>
    </ToolbarContent>
  </Toolbar>
  <AppTable
    cells={
      Array [
        Object {
          "title": "Col1",
        },
      ]
    }
    filtersApplied={false}
    isLoading={false}
    onSort={[MockFunction]}
    rows={
      Array [
        Object {
          "cells": Array [
            "first",
            "second",
            "third",
          ],
        },
      ]
    }
  />
  <SimplePagination
    count={120}
    onChange={[MockFunction]}
    params={
      Object {
        "page": 1,
        "perPage": 10,
      }
    }
  />
</div>
`;

exports[`AppTableWithControls Renders without crashing 1`] = `
<div
  style={
    Object {
      "backgroundColor": "var(--pf-global--BackgroundColor--100)",
    }
  }
>
  <Toolbar
    className="pf-m-toggle-group-container"
    clearFiltersButtonText="actions.clearAllFilters"
    collapseListedFiltersBreakpoint="xl"
  >
    <ToolbarContent
      isExpanded={false}
      showClearFiltersButton={false}
    >
      <ToolbarItem
        alignment={
          Object {
            "default": "alignRight",
          }
        }
        variant="pagination"
      >
        <SimplePagination
          count={120}
          isTop={true}
          onChange={[MockFunction]}
          params={
            Object {
              "page": 1,
              "perPage": 10,
            }
          }
        />
      </ToolbarItem>
    </ToolbarContent>
  </Toolbar>
  <AppTable
    cells={
      Array [
        Object {
          "title": "Col1",
        },
      ]
    }
    filtersApplied={false}
    isLoading={false}
    onSort={[MockFunction]}
    rows={
      Array [
        Object {
          "cells": Array [
            "first",
            "second",
            "third",
          ],
        },
      ]
    }
  />
  <SimplePagination
    count={120}
    onChange={[MockFunction]}
    params={
      Object {
        "page": 1,
        "perPage": 10,
      }
    }
  />
</div>
`;
